samcli/commands/pipeline/bootstrap/cli.py

```diff
@@ -292,6 +320,13 @@ def do_cli(
                 "deployment-branch": deployment_branch,
             }
             pipeline_oidc_provider = GitHubOidcProvider(github_oidc_params, common_oidc_params, GITHUB_ACTIONS)
+        elif oidc_provider == GITLAB:
+            gitlab_oidc_params: dict = {
+                "gitlab-project": gitlab_project,
+                "gitlab-group": gitlab_group,
+                "deployment-branch": deployment_branch,
+            }
```

Comment:
Thinking if we could re-use the logic here. The `deployment_branch` is the same. `github-org` seems match to `gitlab-group` and `github_repo` seems match to `gitlab-project`. Maybe we could use general parameters name for all?

Reply:
GitHub and GitLab are close enough to each other in information that we need that it would probably be fine for those two, but with Bitbucket we will need only a single new parameter, instead of the three that GitHub and GitLab need, the repository UUID which is a bitbucket specific value so I don't think it would be possible to generalize the parameters once we add Bitbucket and if in the future we decide to add any other providers. I think keeping them all separate helps reinforce that to anybody whos reading the code for the first time and is unfamiliar with the concepts